src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/factory/PersistedFactory.kt

```diff
@@ -3,7 +3,9 @@ package uk.gov.justice.digital.hmpps.approvedpremisesapi.factory
 import io.github.bluegroundltd.kfactory.Factory
 import org.springframework.data.jpa.repository.JpaRepository
 
-abstract class PersistedFactory<EntityType : Any, PrimaryKeyType : Any>(private val repository: JpaRepository<EntityType, PrimaryKeyType>) : Factory<EntityType> {
-  fun produceAndPersist(): EntityType = repository.saveAndFlush(this.produce())
+abstract class PersistedFactory<EntityType : Any, PrimaryKeyType : Any>(private val repository: JpaRepository<EntityType, PrimaryKeyType>?) : Factory<EntityType> {
```

Comment:
If this is called `PersistedFactory`, is there an argument for having a `Factory` or `NonPersistedFactory` class instead?